### PR TITLE
Suisin/upm828/cancel phone verification dialog

### PR DESCRIPTION
**Pull request by suisin-deriv**
## Changes:

Add cancel phone verification modal.

### Screenshots:

Screenshot 2024-05-03 at 1 21 02 PM

### CODE REVIEW

**Review by matin-deriv — APPROVED** [review]
LGTM

File: packages/account/src/Sections/Profile/PhoneVerification/__test__/phone-verification-page.spec.tsx [hunk 1]
```diff
@@ -1,14 +1,23 @@
 import { render, screen } from '@testing-library/react';
 import React from 'react';
 import PhoneVerificationPage from '../phone-verification-page';
+import userEvent from '@testing-library/user-event';
```
Comment (yauheni-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/__test__/phone-verification-page.spec.tsx:4]: please check the order for imports 🙏

File: packages/account/src/Sections/Profile/PhoneVerification/__test__/cancel-phone-verification-modal.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    it('it should render CancelPhoneVerificationModal', () => {
+        render(
+            <CancelPhoneVerificationModal
+                show_cancel_verification_modal
+                setShowCancelVerificationModal={mockSetShowCancelModal}
+            />
+        );
+        expect(screen.getByText(/Cancel phone number verification?/)).toBeInTheDocument();
+        expect(screen.getByText(/All details entered will be lost./)).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /Go back/ })).toBeInTheDocument();
```
Comment (yauheni-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/__test__/cancel-phone-verification-modal.spec.tsx:37]: let's move repeated text in separate variables

File: packages/account/src/Sections/Profile/PhoneVerification/__test__/cancel-phone-verification-modal.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import CancelPhoneVerificationModal from '../cancel-phone-verification-modal';
```
Comment (yauheni-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/__test__/cancel-phone-verification-modal.spec.tsx:2]: check here the order of imports please

File: packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import { Button, Text } from '@deriv-com/quill-ui';
+import { Modal } from '@deriv/components';
+import { Localize } from '@deriv/translations';
+import { useHistory } from 'react-router';
```
Comment (yauheni-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx:4]: and here the order please

File: packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import React from 'react';
+
+type TCancelPhoneVerificationModal = {
+    show_cancel_verification_modal: boolean;
```
Comment (yauheni-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx:8]: would it be better to crate more evident got boolean value name, something like should_show_.....
you can ignore this comment :)

File: packages/account/src/Sections/Profile/PhoneVerification/phone-verification.scss [hunk 6]
```diff
@@ -1,4 +1,19 @@
 .phone-verification {
```
Comment (yauheni-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/phone-verification.scss:1]: let's create a variable 
@mixin flex-column {
    display: flex;
    flex-direction: column;
}

and reuse it in many places:

File: packages/account/src/Sections/Profile/PhoneVerification/phone-verification.scss [hunk 7]
```diff
@@ -1,4 +1,19 @@
...
+    &__cancel-modal {
+        &--contents {
+            display: flex;
+            flex-direction: column;
+            gap: 3.2rem;
+        }
+        &--buttons {
+            display: flex;
+            flex-direction: column;
```
Comment (yauheni-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/phone-verification.scss:10]:
```suggestion
@include flex-column;
```

File: packages/account/src/Sections/Profile/PhoneVerification/phone-verification.scss (hunk 6 above)
Comment (suisin-deriv) [reply packages/account/src/Sections/Profile/PhoneVerification/phone-verification.scss:1]: I think if we do that, we need to implement in multiple places as well right 🤔 not just my phone-verification.scss have display: flex; flex-direction: column

**Review by yauheni-deriv — APPROVED** [review]

File: packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+                        <Localize i18n_default_text='Cancel phone number verification?' />
+                    </Text>
+                    <Text>
+                        <Localize i18n_default_text='All details entered will be lost.' />
+                    </Text>
+                </div>
+            </Modal.Body>
+            <Modal.Footer>
+                <div className='phone-verification__cancel-modal--buttons'>
+                    <Button
```
Comment (amina-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx:36]: can we use FormSubmitButton component

File: packages/account/src/Sections/Profile/PhoneVerification/phone-verification-page.tsx [hunk 9]
```diff
@@ -4,23 +4,27 @@ import { LabelPairedArrowLeftCaptionFillIcon } from '@deriv/quill-icons';
...
         <div>
+            <CancelPhoneVerificationModal
+                should_show_cancel_verification_modal={should_show_cancel_verification_modal}
+                setShouldShowCancelVerificationModal={setShouldShowCancelVerificationModal}
+            />
             <div className='phone-verification__redirect_button'>
                 <LabelPairedArrowLeftCaptionFillIcon
                     width={24}
                     height={24}
+                    data-testid='dt-phone-verification-back-btn'
```
Comment (amina-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/phone-verification-page.tsx:27]: data-testid='dt_phone_verification_back_btn'
Testid should be snake case https://wikijs.deriv.cloud/en/Frontend/development-guidelines/test_guide

File: packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx (hunk 8 above)
Comment (suisin-deriv) [reply packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx:36]: I thought we are using button components from quill-ui 🤔

Comment (amina-deriv) [reply packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx:36]: oh yea... Forgot it
Thanks for checking @suisin-deriv

**Review by amina-deriv — APPROVED** [review]

**Pull request merged**